### PR TITLE
`iotedge check` update

### PR DESCRIPTION
**Pull request by daprilik**
Updates `iotedge check` to work with the new underlying `aziot` architecture.

This PR builds on-top of the functionality introduced in the downstream `aziot` PR at https://github.com/Azure/iot-identity-service/pull/83

* * *

In a nutshell, many of the checks that used to be in `iotedge check` have been ported over into the new `aziot check` tool. Please see the associated downstream PR for more details. That being said, we don't want to make users invoke multiple `check` commands, so this PR also includes functionality whereby `iotedge` will shell-out to `aziot` when running `check` and `check-list`, transparently displaying `aziot`'s check results alongside `iotedge`-specific checks.

As with the downstream PR, this PR should be reviewed along two axes:

1. Refactoring work around the actual `iotedge check/check-list` subcommands
2. The actual checks themselves (i.e: confirming that all removed checks have indeed been migrated / aren't applicable anymore).

I would **really appreciate it** if the reviewers actually pulled these changes locally and tested them out, as subtle things such as check output formatting and/or checking check pass/failure cases aren't easy to spot by looking at the code alone.

### CODE REVIEW

File: edgelet/iotedge/src/check/checks/storage_mounted_from_host.rs
Comment (arsing) [inline edgelet/iotedge/src/check/checks/storage_mounted_from_host.rs:98]: Keep the rest of the original comment. It applies to both Windows and Unix.

File: edgelet/iotedge/src/main.rs
Comment (arsing) [inline edgelet/iotedge/src/main.rs:64]: Why does this need to be configurable? Just for local dev? If so, might be useful to make it `#[cfg(debug_assertions)]`

File: edgelet/iotedge/src/check/mod.rs
Comment (arsing) [inline edgelet/iotedge/src/check/mod.rs:225]: `check-list` ought to be changed to be consistent with `check` and use `--output json` instead of `--json`

File: edgelet/iotedge/src/check/mod.rs
Comment (arsing) [inline edgelet/iotedge/src/check/mod.rs:565]: Put the `Err` arm after the `Ok` arm please, to be consistent with the rest of the codebase.

File: edgelet/iotedge/src/check/mod.rs
Comment (arsing) [inline edgelet/iotedge/src/check/mod.rs:229]: Put the `Err` arm after the `Ok` arm please, to be consistent with the rest of the codebase.

**Review by arsing — CHANGES_REQUESTED** [review]

File: edgelet/iotedge/src/main.rs
Comment (daprilik) [reply edgelet/iotedge/src/main.rs:64]: Yes, I've primarily been using it for local dev. I don't imagine any end users would have to manually set it.

Instead of customizing it via CLI flag, how about I read it from an `AZIOT_BIN` env var instead? In that case, should I hide it behind `#[cfg(debug_assertions)]`, or would it be fine to leave it active in release builds as well?

File: edgelet/iotedge/src/check/mod.rs
Comment (daprilik) [reply edgelet/iotedge/src/check/mod.rs:225]: Fair enough, though given that this isn't a flag that users will ever be expected to use directly, I don't think this bit of inconsistency is too big of a deal. Plus, it'll be a bit of a PITA to update `aziot check` now that the first PR has already been merged :wink:

File: edgelet/iotedge/src/main.rs
Comment (daprilik) [reply edgelet/iotedge/src/main.rs:64]: Oh, better yet, I can just use [`option_env!`](https://doc.rust-lang.org/std/macro.option_env.html) instead. We already use it for `IOTEDGE_HOST`. No `#[cfg(debug_assertions)]` needed :smile: 

Sidenote, I didn't know `option_env!` was a thing - very cool!

File: edgelet/iotedge/src/check/mod.rs
Comment (arsing) [reply edgelet/iotedge/src/check/mod.rs]: Yeah, sorry about that.

Comment (daprilik) [reply edgelet/iotedge/src/check/mod.rs]: Sooooo I _should_ make the change then :smile:

Comment (arsing) [reply edgelet/iotedge/src/check/mod.rs]: Yes.

**Review by massand — DISMISSED** [review]

**Comment by massand** [comment]
:shipit:

**Comment by massand** [comment]
Hold off on any submodule update before merge because I might have exposed a bug in the module runtime with the latest aziot version. I'll post a fix for it shortly. If you aren't updating the `aziot` submodule, you should be fine.

**Comment by massand** [comment]
#4240 fixes the above issue. If you don't update the submodule, I can do so in my PR.

**Comment by daprilik** [comment]
> Hold off on any submodule update before merge because I might have exposed a bug in the module runtime with the latest aziot version. I'll post a fix for it shortly. If you aren't updating the `aziot` submodule, you should be fine.

I still need to fix https://github.com/Azure/iotedge/pull/4153#discussion_r555611080, so I'll end up bumping the submodule regardless.

**Comment by massand** [comment]
One note on testing these `iotedge check` changes locally (@micahl can provide guidance): `iotedge check` compares installed version with the latest versions [here](https://github.com/Azure/azure-iotedge/blame/master/latest-versions.json). We should coordinate on adding `aziot-identityd` and adding the version check in this PR perhaps?

**Comment by arsing** [comment]
https://github.com/Azure/iot-identity-service/blob/main/aziot/src/internal/check/checks/mod.rs#L34-L35
________________________________

**Review by huguesBouvier — CHANGES_REQUESTED** [review]
Should be tested in nested configuration. Check it out with me

File: edgelet/iotedge/src/init/import/mod.rs
Comment (arsing) [inline edgelet/iotedge/src/init/import/mod.rs:271]: No. `always_reprovision_on_startup` and `dynamic_reprovisioning` are not the same thing.

`always_reprovision_on_startup` needs to be set to what it was set in the old config.

Comment (daprilik) [reply edgelet/iotedge/src/init/import/mod.rs:271]: @massand said this is the fix :cry:

Comment (arsing) [reply edgelet/iotedge/src/init/import/mod.rs:271]: I don't know why he said that, but it doesn't change what I said.

Comment (daprilik) [reply edgelet/iotedge/src/init/import/mod.rs:271]: Wait, so what should the fix be for the `ProvisioningType::Manual` cases?

Comment (daprilik) [reply edgelet/iotedge/src/init/import/mod.rs:271]: @arsing

Comment (daprilik) [reply edgelet/iotedge/src/init/import/mod.rs:271]: Also, on that note, it looks like I need to change the test-files as well.

I can't say I'm a fan of how these changes are getting bundled in with this unrelated PR, but if it's happening, I could use some guidance on what to do.

Comment (arsing) [reply edgelet/iotedge/src/init/import/mod.rs:271]: https://github.com/Azure/iotedge/blob/release/1.0.10/edgelet/edgelet-core/src/settings.rs#L426-L430

Comment (daprilik) [reply edgelet/iotedge/src/init/import/mod.rs:271]: Gotcha. So I'll just set them to true then. Cool.

Comment (daprilik) [reply edgelet/iotedge/src/init/import/mod.rs]: It is done

File: edgelet/iotedge/src/init/import/mod.rs
Comment (arsing) [inline edgelet/iotedge/src/init/import/mod.rs:256]: Add a comment on this line along the lines of `// TODO: Migrate this to edged config when support for dynamic reprovisioning is reinstated in edged.`

@massand Do you have a work item for it already?

**Review by arsing — DISMISSED** [review]

File: edgelet/iotedge/src/init/import/mod.rs
Comment (daprilik) [reply edgelet/iotedge/src/init/import/mod.rs:257]: added the comment

**Review by huguesBouvier — CHANGES_REQUESTED** [review]
Best careful, the run on the pipeline is not compiling:
 use edgelet_docker::{Settings, UPSTREAM_PARENT_KEYWORD};
  |                                ^^^^^^^^^^^^^^^^^^^^^^^ no `UPSTREAM_PARENT_KEYWORD` in the root

**Comment by daprilik** [comment]
> Best careful, the run on the pipeline is not compiling:
> use edgelet_docker::{Settings, UPSTREAM_PARENT_KEYWORD};
> | ^^^^^^^^^^^^^^^^^^^^^^^ no `UPSTREAM_PARENT_KEYWORD` in the root

Yes indeed, I realized that and force pushed a fix.

**Review by huguesBouvier — APPROVED** [review]

**Review by arsing — APPROVED** [review]

**Review by massand — APPROVED** [review]
I had hit some check errors due to diagnostics module misbehaving. But, the other checks seemed fine. Let us test this after merge, since it will be easier. Thanks!

**Pull request merged**
